### DISCUSSION TITLE
migrate from g6 to g7 requires reinstall

### DISCUSSION THREAD

**Discussion by plickplick** [NightscoutFoundation/xDrip#3321] [Bug reports]
Not able to connect to g7 from my Google Pixel 7 pro.

I followed the instructions from navid200.
Btw Exelent instructions!!
(Making sure all settings were correct, updating to latest nightly build)

I just could not get a connection.

After several hours of trying I gave up and uninstalled xdrip, installed xdrip again. And it worked!

I found multiple posts where people have had the same experience with different issues not only connecting. My  conclusion is updating is just not enough. Due to corrupted updating or old broken settings. I suggest the instructions you have should be changed to reinstall instead of updating xdrip. This would save a lot of people headaches.

Thank you all for an Exelent application and Exelent support!!! ❤️❤️

**Comment by Navid200** [comment]
The only time it may be beneficial to uninstall xDrip is if something has been done to interfere with xDrip pairing.
For example, if you already have used Dexcom G7 app to connect, switching to xDrip without forgetting the device may delay the connectivity.  It should eventually connect.  But, it is best to forget the device before entering the transmitter ID in xDrip.
That has been my experience.

**Reply by plickplick** [reply]
I get that this is your experience but it was not mine. I followed all your instructions. Fully aware of the Bluetooth shortcomings. I did remove all devices had not previously connected to the g7 and so on. Watched in the status page for hours even left it in status page over night. Nothing . .  Figured it could be a faulty g7 unit broke the seal on the dexcom device and connected it. No problems at all. Tried with xdrip again nothin.Uninstalled xdrip installed it again. And everything worked as it should. Got a connection! And there is a lot of other things like restarting the phone restarting Bluetooth that I tried that did not work. Reinstall semes to solve a lot of issues for a lot of people. And it is really easy to reinstall no point in avoiding it.

**Reply by Navid200** [reply]
Unless you don't want to lose all your previous readings and all your customized settings in xDrip.  Then, if you uninstall without first backing them up, it will be very inconvenient.  That's why I always tell people to uninstall as a last resort after telling them that they will lose their old readings and settings.

That's why I don't want to tell everyone to do that by default because everyone has not had your experience.

**Reply by plickplick** [reply]
A good point!

That is why I sat for hours without unistalling. 

Anyways I really appreciate all your hard work and your instructions are Exelent. My suggestion is a side note. And still stands added as a bottom line with your warnings about backing up and so on would help. 

Or maybe someone should look into the update process. This is not the first time I have had issues updating with several different phones. But a programmer myself I know how much work that could be so not really suggesting it. Just a thought.
